### PR TITLE
feat: add maubot rock

### PR DESCRIPTION
**Pull request by amandahla**
Applicable spec: <link>

### Overview

Maubot is a python application. This PR creates a rock for it.

Reference:
https://docs.mau.fi/maubot/usage/setup/index.html

### Rationale

Create a rock to be published as a charm resource.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

No CH available yet.

### CODE REVIEW

File: maubot_rock/rockcraft.yaml [hunk 1]
```diff
@@ -4,16 +4,29 @@
...
+environment: 
+  PYTHONPATH: /opt/maubot/python/dist
+
 parts:
-  hello:
+  maubot:
     plugin: nil
-    stage-packages:
-      - hello
+    build-packages:
```
Comment (arturo-seijas) [inline maubot_rock/rockcraft.yaml:21]: Have you tried using the[ python plugin](https://documentation.ubuntu.com/rockcraft/en/latest/reference/plugins/python_plugin/)?

**Review by arturo-seijas — DISMISSED** [review]

File: maubot_rock/rockcraft.yaml (hunk 1 above)
Comment (amandahla) [reply maubot_rock/rockcraft.yaml]: Done :)

**Review by arturo-seijas — DISMISSED** [review]

File: maubot_rock/rockcraft.yaml [hunk 2]
```diff
@@ -4,16 +4,22 @@
 name: maubot
 summary: Maubot rock
 description: Maubot OCI image for the Maubot charm
-version: "1.0"
+version: "0.5.0"
```
Comment (nrobinaubertin) [inline maubot_rock/rockcraft.yaml:7]: You used the maubot version here, shouldn't it instead reflect the version of the rock ?

Comment (amandahla) [reply maubot_rock/rockcraft.yaml:7]: I got inspired by the https://github.com/canonical/prometheus-rock
Although it is not defined in any of your specs, I think it makes sense that way. WDYT?
@gregory-schiano thoughts?

**Review by nrobinaubertin — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

File: maubot_rock/rockcraft.yaml (hunk 2 above)
Comment (amandahla) [reply maubot_rock/rockcraft.yaml:7]: You have a good point. I'll close this PR but this can be discussed further

**Pull request merged**